lib/logstash/filters/bytes.rb

```diff
@@ -0,0 +1,100 @@
+# encoding: utf-8
+require "logstash/filters/base"
+require "logstash/namespace"
+
+# This filter will parse a given string as a computer storage
+# value (e.g. "123 MB" or "5.3GB") and add a new numeric field with
+# size in bytes
+#
+class LogStash::Filters::Bytes < LogStash::Filters::Base
+
+  PREFIX_POWERS = {
+    'k' => 1, # 1 kilobyte = 1024 ^ 1 bytes
```

Comment:
The prefixes mean different things depending on whether we're talking network data or size on disk.  See: https://en.wikipedia.org/wiki/Mebibyte .

It'd be nice to also support `MiB` and `KiB`, even though their use is a little rare.

More important would be to add an option that toggles between SI Units and powers of 2.

Reply:
By SI units, I mean `k = 1000`